src/common/pf_alarm.c

```diff
@@ -748,11 +699,30 @@ static int pf_alarm_apmr_low_handler (
       p_apmr_msg = &p_apmx->apmr_msg[nbr];
       p_apmr_msg->p_buf = p_buf;
       p_apmr_msg->frame_id_pos = frame_id_pos;
-      if (os_mbox_post (p_apmx->p_alarm_q, (void *)p_apmr_msg, 0) != 0)
+      if (p_apmx->p_alarm_q != NULL)
       {
-         LOG_ERROR (PF_ALARM_LOG, "Alarm(%d): Lost one alarm\n", __LINE__);
-         ret = 0; /* Failed to handle frame. The calling function needs to free
-                     the buffer. */
+         if (os_mbox_post (p_apmx->p_alarm_q, (void *)p_apmr_msg, 0) == 0)
+         {
+            ret = 1; /* Means that calling function should not free buffer,
+                        as that will be done when reading the mbox */
+         }
+         else
+         {
+            LOG_ERROR (
+               PF_ALARM_LOG,
+               "Alarm(%d): Failed to put incoming %s prio alarm in mbox\n",
+               __LINE__,
+               priotext);
+         }
+      }
+      else
+      {
+        LOG_ERROR (
+           PF_ALARM_LOG,
+           "Alarm(%d): Could not put incoming %s prio alarm frame in"
+               " mbox, as it is deallocated.\n",
+            __LINE__,
+            priotext);
       }
    }
```

Comment:
Return 1 when p_buf is NULL (as there is no buffer to free)?

Reply:
Good point.
Done.